aries_cloudagent/protocols/coordinate_mediation/v1_0/routes.py

```diff
@@ -0,0 +1,456 @@
+"""coordinate mediation admin routes."""
+
+# https://github.com/hyperledger/aries-rfcs/tree/master/features/0211-route-coordination#0211-mediator-coordination-protocol
+
+from aiohttp import web
+from aiohttp_apispec import (
+    docs,
+    match_info_schema,
+    querystring_schema,
+    request_schema,
+    response_schema,
+)
+from marshmallow import fields, validate
+
+from ....admin.request_context import AdminRequestContext
+from ....connections.models.conn_record import ConnRecord
+from ....messaging.models.base import BaseModelError
+from ....messaging.models.openapi import OpenAPISchema
+from ....storage.error import StorageError, StorageNotFoundError
+from ...routing.v1_0.models.route_record import RouteRecord, RouteRecordSchema
+from .manager import MediationManager
+from .message_types import SPEC_URI
+from .messages.inner.keylist_update_rule import (
+    KeylistUpdateRule,
+    KeylistUpdateRuleSchema,
+)
+from .messages.keylist_update_response import KeylistUpdateResponseSchema
+from .messages.mediate_deny import MediationDenySchema
+from .messages.mediate_grant import MediationGrantSchema
+from .models.mediation_record import MediationRecord, MediationRecordSchema
+from .models.mediation_schemas import (
+    CONNECTION_ID_SCHEMA,
+    MEDIATION_ID_SCHEMA,
+    MEDIATION_STATE_SCHEMA,
+    MEDIATOR_TERMS_SCHEMA,
+    RECIPIENT_TERMS_SCHEMA,
+)
+
+
+class MediationListSchema(OpenAPISchema):
+    """Result schema for mediation list query."""
+
+    results = fields.List(
+        fields.Nested(MediationRecordSchema),
+        description="List of mediation records",
+    )
+
+
+class MediationListQueryStringSchema(OpenAPISchema):
+    """Parameters and validators for mediation record list request query string."""
+
+    conn_id = CONNECTION_ID_SCHEMA
+    mediator_terms = MEDIATOR_TERMS_SCHEMA
+    recipient_terms = RECIPIENT_TERMS_SCHEMA
+    state = MEDIATION_STATE_SCHEMA
+
+
+class MediationCreateRequestSchema(OpenAPISchema):
+    """Parameters and validators for create Mediation request query string."""
+
+    mediator_terms = MEDIATOR_TERMS_SCHEMA
+    recipient_terms = RECIPIENT_TERMS_SCHEMA
+
+
+class MediationCreateRequestQuerySchema(OpenAPISchema):
+    """Create mediation request query string schema."""
+
+    conn_id = CONNECTION_ID_SCHEMA
```

Comment:
`CONNECTION_ID_SCHEMA` marks it as optional but it should be required in this case. Given that, maybe we should consider making this a path match var again.

Reply:
CONNECTION_ID_SCHEMA appears to be pulled in from the mediation protocol so we can change it without messing up other admin API's. We can also create a new field in place easily if required is all that's needed.
something like,
`fields.UUID( 
    description="Connection identifier (required)",
    required=True,
    example=UUIDFour.EXAMPLE,
)`
or possibly dynamically change the required key: value inside the CONNECTION_ID_SCHEMA.
I have no preference for how the API should look, path match var or body match var.